R/coercion.R

```diff
@@ -0,0 +1,500 @@
+#' [as.function()] method for `<epidist>` class
+#'
+#' @description Converts an `<epidist>` object to a distribution function
+#' (see [epidist_distribution_functions]), either probability density/mass
+#' function, (`density`), cumulative distribution function (`cdf`), random
+#' number generator (`generate`), or quantile (`quantile`).
+#'
+#' @details The function returned takes a single required argument `x`.
+#'
+#' @inheritParams print.epidist
+#' @inheritParams base::as.function
+#' @param func_type A single `character` string specifying which distribution to
+#' convert `<epidist>` object into. Default is `"density"`. Other options are
+#' `"cdf"`, `"generate"`, or `"quantile"`.
+#'
+#' @return A [function] object.
+#' @export
+as.function.epidist <- function(x,
+                                func_type = c(
+                                  "density", "cdf", "generate", "quantile"
+                                ),
+                                ...) {
+  chkDots(...)
+  func_type <- match.arg(func_type)
+  if (!is_parameterised(x)) {
+    stop(
+      "Cannot convert unparameterised <epidist> to distribution function",
+      call. = FALSE
+    )
+  }
+  epidist <- x
+  x <- switch(func_type,
+    density = function(x) density(x = epidist, at = x),
+    cdf = function(x) epiparameter::cdf(x = epidist, q = x),
+    generate = function(x) epiparameter::generate(x = epidist, times = x),
+    quantile = function(x) quantile(x = epidist, p = x)
+  )
+
+  # return distribution function
+  x
```

Comment:
```suggestion
  # return distribution function
  switch(func_type,
    density = function(x) density(x = epidist, at = x),
    cdf = function(x) epiparameter::cdf(x = epidist, q = x),
    generate = function(x) epiparameter::generate(x = epidist, times = x),
    quantile = function(x) quantile(x = epidist, p = x)
  )
```

Too many things are named x otherwise and it's a bit hard to read.